azurelinuxagent/common/persist_firewall_rules.py

```diff
@@ -0,0 +1,219 @@
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+import os
+import sys
+
+import shutil
+
+from azurelinuxagent.common import logger
+from azurelinuxagent.common.cgroupapi import CGroupsApi
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.osutil import get_osutil
+from azurelinuxagent.common.utils import shellutil, fileutil
+from azurelinuxagent.common.utils.networkutil import AddFirewallRules
+from azurelinuxagent.common.utils.shellutil import CommandError
+
+
+class PersistFirewallRulesHandler(object):
+
+    __SERVICE_FILE_CONTENT = """
+# This drop-in unit file was created by the Azure VM Agent.
+# Do not edit.
+[Unit]
+Description=Setup network rules for WALinuxAgent 
+Before=network-pre.target
+Wants=network-pre.target
+DefaultDependencies=no
+ConditionPathExists={0}
+
+[Service]
+Type=oneshot
+Environment="DST_IP={1}" "UID={2}" "WAIT={3}"
+ExecStart={4} {0} --dst_ip=${{DST_IP}} --uid=${{UID}} ${{WAIT}}
+RemainAfterExit=false
+
+[Install]
+WantedBy=network.target
+"""
+
+    __OVERRIDE_CONTENT = """
+# This drop-in unit file was created by the Azure VM Agent.
+# Do not edit.
+[Service]
+Environment=
+Environment="DST_IP={0}" "UID={1}" "WAIT={2}"
+"""
+
+    _AGENT_NETWORK_SETUP_BIN_FILE = "waagent_network_setup.py"
+    _AGENT_NETWORK_SETUP_NAME_FORMAT = "{0}-network-setup.service"
+    _DROP_IN_ENV_FILE_NAME = "10-environment.conf"
+
+    _FIREWALLD_RUNNING_CMD = ["firewall-cmd", "--state"]
+
+    def __init__(self, dst_ip, uid):
+        """
+        If not, then create the new service file and copy it over to the /lib/systemd/system/ if doesnt exist
```

Comment:
"If not," incomplete sentence?

Reply:
Ohh shoot sorry forgot to re-write this comment. Will do!